subscriptions/restHook.ts

```diff
@@ -0,0 +1,127 @@
+import axios from 'axios';
+import { SubscriptionEndpoint } from 'fhir-works-on-aws-routing/lib/router/validation/subscriptionValidator';
+import { groupBy } from 'lodash';
+import { makeLogger } from 'fhir-works-on-aws-interface';
+import { SubscriptionMatchMessage } from './types';
+import getAllowListedSubscriptionEndpoints from './allowList';
+import ensureAsyncInit from '../src/index';
+
+const logger = makeLogger();
+
+const SINGLE_TENANT_ALLOW_LIST_KEY = 'SINGLE_TENANT_ALLOW_LIST_KEY';
+
+export interface AllowListInfo {
```

Comment:
can you move all the functions related to the allow list to a separate file? I feel that this file is bit too big